### PR TITLE
Fix a flaky wget issue.

### PR DESCRIPTION
**Pull request by vanbasten23**
# Description

In b/329297378 and b/329295948, we have seen an error:
```
[2024-03-11, 14:06:30 UTC] {gke.py:116} INFO - pt-nightly-resnet50-mp-fake-v100-x2x2-9jhn2-1-6x5j8] --2024-03-11 14:06:30--  https://github.com/pytorch/xla/archive/refs/heads/master.tar.gz
[2024-03-11, 14:06:50 UTC] {gke.py:116} INFO - pt-nightly-resnet50-mp-fake-v100-x2x2-9jhn2-1-6x5j8] Resolving github.com (github.com)... failed: Temporary failure in name resolution.
[2024-03-11, 14:06:50 UTC] {gke.py:116} INFO - pt-nightly-resnet50-mp-fake-v100-x2x2-9jhn2-1-6x5j8] wget: unable to resolve host address ‘github.com’
[2024-03-11, 14:06:50 UTC] {gke.py:116} INFO - pt-nightly-resnet50-mp-fake-v100-x2x2-9jhn2-1-6x5j8] gzip: stdin: unexpected end of file
[2024-03-11, 14:06:50 UTC] {gke.py:116} INFO - pt-nightly-resnet50-mp-fake-v100-x2x2-9jhn2-1-6x5j8] tar: Child returned status 1
[2024-03-11, 14:06:50 UTC] {gke.py:116} INFO - pt-nightly-resnet50-mp-fake-v100-x2x2-9jhn2-1-6x5j8] tar: Error is not recoverable: exiting now
```
for 2 days (03/11 and 03/12). We didn't have seen it before and it recovered again on 03/13. I see 2 options:
- using wget: per this [so](https://unix.stackexchange.com/questions/168584/wget-is-unable-to-resolve-host-address-80-of-the-time), we can either retry on the failure of `wget` or do `wget --inet4-only`
- using `git clone` with `--depth 1` so we don't have to download the whole git history, which is the intention of using `wget` previously.

Option2 seems to be more stable, meanwhile not have to download the whole git history.
@will-cromar  please let me know what you think.

# Tests

Triggered the GPU multi-host tests

**Instruction and/or command lines to reproduce your tests:** ...

**List links for your tests (use go/shortn-gen for any internal link):** https://screenshot.googleplex.com/8uEm7p4TFjKgW49

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/legacy_test/tests/pytorch/nightly/common.libsonnet [hunk 1]
```diff
@@ -141,8 +141,9 @@ local volumes = import 'templates/volumes.libsonnet';
         pip3 install --pre torch torchvision --index-url https://download.pytorch.org/whl/nightly/cpu
 
         mkdir pytorch
-        wget https://github.com/pytorch/xla/archive/refs/heads/master.tar.gz -O - | tar xzf -
-        mv xla-master pytorch/xla
+        cd pytorch
+        git clone --depth=1 https://github.com/pytorch/xla.git
```
Comment (will-cromar) [inline dags/legacy_test/tests/pytorch/nightly/common.libsonnet:145]: Can you clone directly into `pytorch`? `git clone --depth=1 https://github.com/pytorch/xla.git pytorch`

**Review by will-cromar — APPROVED** [review]
Wow, this is an obscure one. Thanks for looking into it. LGTM

File: dags/legacy_test/tests/pytorch/nightly/common.libsonnet (hunk 1 above)
Comment (jonb377) [reply dags/legacy_test/tests/pytorch/nightly/common.libsonnet]: @vanbasten23 I think you'd need to `git clone https://... pytorch/xla`

**Review by jonb377 — APPROVED** [review]
Thanks Xiongfei!

File: dags/legacy_test/tests/pytorch/nightly/common.libsonnet (hunk 1 above)
Comment (vanbasten23) [reply dags/legacy_test/tests/pytorch/nightly/common.libsonnet]: yes!

**Comment by vanbasten23** [comment]
Thanks for the review!

**Pull request merged**